src/helium_follower_service.erl

```diff
@@ -54,8 +55,50 @@ subnetwork_last_reward_height(Ctx, Req) ->
 
 -spec active_gateways(follower_pb:follower_gateway_stream_req_v1_pb(), grpcbox_stream:t()) ->
         {ok, grpcbox_stream:t()} | grpcbox_stream:grpc_error_response().
+active_gateways(#follower_gateway_stream_req_v1_pb{} = _Msg, StreamState) ->
+    #{chain := Chain} =  grpcbox_stream:stream_handler_state(StreamState),
+    case Chain of
+        undefined ->
+            lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+            {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+        _ ->
+            grpcbox_stream:update_headers([{<<"gateway_stream">>, <<"start">>}], StreamState),
```

Comment:
you will need to handle the returned State here and use that from this point forward